python/lsst/cp/pipe/bfCoeffCalc.py

```diff
@@ -0,0 +1,1272 @@
+        default='ccd',
+    )
+    maxIterRegression = pexConfig.Field(
+        dtype=int,
+        doc="Maximum number of iterations for the iterative regression fitter",
+        default=10
+    )
+    nSigmaClipGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="Number of sigma to clip to during gain calculation",
+        default=5
+    )
+    nSigmaClipRegression = pexConfig.Field(
+        dtype=int,
+        doc="Number of sigma to clip to during iterative regression",
+        default=3
+    )
+    xcorrCheckRejectLevel = pexConfig.Field(
+        dtype=float,
+        doc="Sanity check level for the sum of the input cross-correlations. Arrays which "
+        "sum to greater than this are discarded before the clipped mean is calculated.",
+        default=1.0  # xxx change this back once the problem is fixed!
+        # default=0.2
+    )
+    maxIterSOR = pexConfig.Field(
+        dtype=int,
+        doc="The maximum number of iterations allowed for the successive over-relaxation method",
+        default=10000
+    )
+    eLevelSOR = pexConfig.Field(
+        dtype=float,
+        doc="The target residual error for the successive over-relaxation method",
+        default=5.0e-14
+    )
+    nSigmaClipKernelGen = pexConfig.Field(
+        dtype=float,
+        doc="Number of sigma to clip to during pixel-wise clipping when generating the kernel",
+        default=4
+    )
+    nSigmaClipXCorr = pexConfig.Field(
+        dtype=float,
+        doc="Number of sigma to clip when calculating means for the cross correlation",
+        default=5
+    )
+    maxLag = pexConfig.Field(
+        dtype=int,
+        doc="The maximum lag to use when calculating the cross correlation/kernel",
+        default=8
+    )
+    nPixBorderGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the gain",
+        default=10
+    )
+    nPixBorderXCorr = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the cross correlation/kernel",
+        default=10
+    )
+    biasCorr = pexConfig.Field(
+        dtype=float,
+        doc="A scary, empirically determined correction-factor correcting for sigma-clipping" +
+        " a non-Gaussian distribution",
+        default=0.9241
+    )
+    backgroundBinSize = pexConfig.Field(
+        dtype=int,
+        doc="Size of the background bins",
+        default=128
+    )
+    fixPtcThroughOrigin = pexConfig.Field(
+        dtype=bool,
+        doc="Contrain the fit of the PTC to go through the origin?",
+        default=True
+    )
+    level = pexConfig.ChoiceField(
+        doc="The level at which to calculate kernels",
+        dtype=str, default="CCD",
+        allowed={
+            "AMP": "Every amplifier treated separately",
+            "CCD": "One kernel per CCD",
+        }
+    )
+
+    def validateIsrConfig(self, logger):
```

Comment:
If there are parts that can usefully validate the config without logging then they should probably be moved to `validate` (or a method called by `validate` in the config. But otherwise your solution seems best and it will certainly suffice.

Reply:
Yeah, I'm torn. It's not _really_ validation is the thing; they're not intrinsically invalid, they're just ensuring that people aren't Doing It Wrong™️, but they're not testing for a non-compliant pexConfig, so I think it should probably stay as-is really.

Reply:
I'd say if it raises an exception then it's validation. We do this sort of thing in other places -- check that something was processed with required config settings before being allowed to proceed -- and we call it validation there.

However, I am not saying you have to split this into `xConfig.validate` (with no logging) and the rest of it. If it feels like a coherent bit of code to you then it's probably best to leave as a single method.

Reply:
OK, I've just renamed the method to `validateIsrConfig()` config then, and left it as it was (after the changes I made before of course, so it still lives in the main class).